dll/cpl/sysdm/environment.c

```diff
@@ -472,7 +479,238 @@ OnDeleteVariable(HWND hwndDlg,
     }
 }
 
+static VOID
+OnEnvironmentDlgResize(HWND hwndDlg,
+                       PENVIRONMENT_DIALOG_DATA DlgData,
+                       DWORD cx,
+                       DWORD cy)
+{
+    RECT rect;
+    INT Colx, y = 0;
+    HDWP hdwp = NULL;
+    HWND hItemWnd;
+
+    if ((cx != DlgData->cxOld) || (cy != DlgData->cyOld))
+    {
+        hdwp = BeginDeferWindowPos(13);
+
+        if (cy >= DlgData->cyOld)
+            y += (cy - DlgData->cyOld) / 2 + (cy - DlgData->cyOld) % 2;
```

Comment:
This might probably be simplified to:
```suggestion
            y += (cy - DlgData->cyOld + 1) / 2;
```
(do it at your will).

Proof, using integer arithmetics:
- suppose `cy - DlgData->cyOld` is even (`== 2*k` for some `k`), then:
  `(cy - DlgData->cyOld + 1) / 2 == (2*k + 1) / 2 == k`, since the integer division rounds down.
  Compare with: `(cy - DlgData->cyOld) / 2 + (cy - DlgData->cyOld) % 2 == (2*k)/2 + (2*k)%2 == k + 0 == k`.
- suppose instead  `cy - DlgData->cyOld` is odd (`== 2*k+1` for some `k`), then:
  `(cy - DlgData->cyOld + 1) / 2 == (2*k+1 + 1) / 2 == (2*k + 2) / 2 == k+1`.
  Compare with: `(cy - DlgData->cyOld) / 2 + (cy - DlgData->cyOld) % 2 == (2*k+1)/2 + (2*k+1)%2 == k + 1`.
QED.
This proof supposes that there is no wrap-up in the evaluation of `cy - DlgData->cyOld + 1`, i.e. that the difference `cy - DlgData->cyOld` cannot be larger than maximum integer value.